distributed/worker_memory.py

```diff
@@ -161,18 +160,12 @@ async def memory_monitor(self, worker: Worker) -> None:
         If process memory rises above the pause threshold (80%), stop execution of new
         tasks.
         """
-        if self._memory_monitoring:
```

Comment:
Nit: Let's remove `_memory_monitoring` everywhere since it is not being used anymore.